recipe/meta.yaml

```diff
@@ -1,20 +1,30 @@
-{% set version = "1.13.1" %}
+{% set version = "1.14.1" %}
 
 package:
   name: scipy
   version: {{ version }}
 
 source:
   - url: https://github.com/scipy/scipy/releases/download/v{{ version }}/scipy-{{ version }}.tar.gz
-    sha256: 095a87a0312b08dfd6a6155cbbd310a8c51800fc931b8c0b84003014b874ed3c
+    sha256: 5a275584e726026a5699459aa72f828a610821006228e841b94275c4a7c08417
     patches:                                                     # [s390x]
       - patches/0002-arff-nodata-test-remove-endian-check.patch  # [s390x]
 
 build:
   number: 0
-  skip: true  # [py<39]
-  missing_dso_whitelist:  # [linux and s390x]
-    - $RPATH/ld64.so.1    # [linux and s390x] 
+  skip: true  # [py<310]
+  missing_dso_whitelist:
+    - $RPATH/ld64.so.1  # [linux and s390x]
+    # sf_error_state is loaded with ctypes.WinDLL
+    # see: https://github.com/scipy/scipy/pull/20321
+    - "**/sf_error_state.dll" # [win]
+    # It is also flagged as overlinking on linux, though this seems to be a conda issue.
+    # For some reason, the .sos built for python 3.11 look for .sos for linking in directories of a different
+    # python. This patches over this, although it's not great and the problem should be fixed properly.
+    # Note that this issue is sporadic - you may remove this, and the build may work but then break at a later date.
+    # These are present in the site-packages directory
```

Comment:
Lines 22-25 are copy-pasted from the pytorch recipe. fyi, the reason for the entries there being needed is actually because the .sos are manually loaded by the pytorch python code, rather than using the RPATH in the binaries to find their dependencies. Not sure if this is relevant here

Reply:
I believe that is. Happy to be proven otherwise.

Reply:
I mean: here the dso whitelist entries are required, but this comment is quite vague about the reason. The reason in the pytorch recipe - the original source of the now-outdated comment - is because the python code loads the dsos and the RPATH entries aren't used. Perhaps that's the reason here as well, in which case the comment can be updated.